test/helper/server_check_helper_test.rb

```diff
@@ -0,0 +1,51 @@
+require 'test/unit'
+require_relative '../../recipes/server_check_helper'
+
+class TestServerCheck < Test::Unit::TestCase
+  def setup
+    @tests = [
+      {
+        'retry_join_list' => nil,
+        'my_ip' => '10.10.10.10',
+        'disallowed' => false,
```

Comment:
Personal preference - this is a double negative so a little hard to parse quickly.

```suggestion
        'allowed' => true,
```